### PR TITLE
[data][doc] Add DatasetConfig -> DataConfig migration guide 

### LINKED ISSUE #35958: [data] [docs] Update docs to reflect new Data+Train API
After https://github.com/ray-project/ray/issues/32605, we need to update the examples in the docs to use the new API. Specifically:

1. We should no longer have any examples using the `preprocessor` arg for Trainer.
2. We should no longer have any examples use the `DatasetConfig` class (they should use DataConfig instead).

cc @raulchen @bveeramani

### PR DESCRIPTION
**Pull request by raulchen**
## Why are these changes needed?

- Add DatasetConfig -> DataConfig migration guide
- Move `ray.train.data_config` to `ray.train._internal.data_config`

## Related issue number

closes #35958

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — APPROVED** [review]

File: doc/source/ray-air/doc_code/air_ingest_migration.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        "test": DatasetConfig(
+            split=False,
+        ),
+    },
+)
+my_trainer.fit()
+# __legacy_api_end__
+
+# __new_api__
+from ray.train.data_config import DataConfig
```
Comment (pcmoritz) [inline doc/source/ray-air/doc_code/air_ingest_migration.py:40]: We should be clear here that `ray.train.data_config` is not a public API. All the configs should be included via `ray.train` directly going forward. Also see the PR description of https://github.com/ray-project/ray/pull/36706 and the REP https://github.com/ray-project/enhancements/pull/36 :)

Comment (pcmoritz) [reply doc/source/ray-air/doc_code/air_ingest_migration.py:40]: @ericl maybe we should rename `data_config` to `_data_config` so there are not two ways to do this and people get confused?

Comment (ericl) [reply doc/source/ray-air/doc_code/air_ingest_migration.py:40]: We should put it in a _internal folder then probably, to follow convention of other libraries.

Comment (pcmoritz) [reply doc/source/ray-air/doc_code/air_ingest_migration.py:40]: Ok, let's replace `ray.train.data_config` with `ray.train` everywhere in this PR before merging, and then we can do the `_internal` refactor later as time permits (doesn't need to be part of 2.6).

Comment (raulchen) [reply doc/source/ray-air/doc_code/air_ingest_migration.py:40]: So I assume nothing is needed in this PR, right?

Comment (pcmoritz) [reply doc/source/ray-air/doc_code/air_ingest_migration.py:40]: It is -- you need to replace `ray.train.data_config.DataConfig` with `ray.train.DataConfig` everywhere :)

Comment (raulchen) [reply doc/source/ray-air/doc_code/air_ingest_migration.py:40]: oh, I didn't know it's already in `ray.train`. I think there may be other places using `ray.train.data_config` as well. Will update all.

Comment (pcmoritz) [reply doc/source/ray-air/doc_code/air_ingest_migration.py:40]: Thanks, that's awesome :)

**Review by pcmoritz — APPROVED** [review]
Nice, feel free to merge if the CI passes :)

**Comment by raulchen** [comment]
@pcmoritz also moved `data_config` to `_internal`. Does this PR need to be cherry-picked to 2.6.0?

**Comment by pcmoritz** [comment]
Thanks, that's great! Yes, it needs to be cherry-picked :)

**Pull request merged**